src/Npgsql/Internal/TypeHandlers/ByteaHandler.cs

```diff
@@ -43,19 +44,44 @@ public override async ValueTask<byte[]> Read(NpgsqlReadBuffer buf, int len, bool
 
     ValueTask<ArraySegment<byte>> INpgsqlTypeHandler<ArraySegment<byte>>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription? fieldDescription)
         => throw new NotSupportedException("Only writing ArraySegment<byte> to PostgreSQL bytea is supported, no reading.");
+    
+    ValueTask<Stream> INpgsqlTypeHandler<Stream>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription? fieldDescription)
+        => throw new NotSupportedException("Only writing Stream to PostgreSQL bytea is supported, no reading.");
 
     int ValidateAndGetLength(int bufferLen, NpgsqlParameter? parameter)
         => parameter == null || parameter.Size <= 0 || parameter.Size >= bufferLen
             ? bufferLen
             : parameter.Size;
 
+    int ValidateAndGetLength(Stream stream, NpgsqlParameter? parameter)
+    {
+        if (parameter != null && parameter.Size > 0)
+            return parameter.Size;
+        
+        if (!stream.CanSeek)
+            throw new NpgsqlException("Cannot write a stream of bytes. Either provide a positive size, or a seekable stream.");
+
+        try
+        {
+            return (int)(stream.Length - stream.Position);
+        }
+        catch (Exception ex)
```

Comment:
For slightly better clarity?

```suggestion
        catch (OverflowException ex)
```

Reply:
Actually, here I was hoping to handle an exception from accessing either `stream.Length` or `stream.Position` (neither of them should throw, but...)

Reply:
Ah missed that.

Yeah, the (exotic) overflow case would be better handled by an up-front arithmetic check, rather than catching the exception. Aside from that yuo're right that we should catch the stream exceptions...